MdeModulePkg/Include/Library/DeviceStateLib.h

```diff
@@ -32,27 +32,85 @@ MU_CHANGE: new file
 
 #define DEVICE_STATE_MAX  (1 << (31))
 
+#define DEVICE_STATE_PLATFORM_MODE_INDEX_START  20
+#define DEVICE_STATE_PLATFORM_MODE_INDEX_END    27
+
+#define MAX_INSECURE_DEVICE_STATE_STRING_SIZE  512
+
+//
+// DEFINE the device state strings here. These are used for logging and measuring purposes.
+//
+#define DEVICE_STATE_SECUREBOOT_OFF_STRING             "SECURE_BOOT_OFF "
+#define DEVICE_STATE_MANUFACTURING_MODE_STRING         "MANUFACTURING_MODE "
+#define DEVICE_STATE_DEVELOPMENT_BUILD_ENABLED_STRING  "DEVELOPMENT_BUILD_ENABLED "
+#define DEVICE_STATE_SOURCE_DEBUG_ENABLED_STRING       "SOURCE_DEBUG_ENABLED "

```

Comment:
I assume in Tcg2Pei you will also measure the specific TCG spec defined debugger string?

Reply:
https://github.com/microsoft/mu_tiano_plus/pull/342/files#diff-dd252bb953ec11695bb497d99150fe3c48e069945f08a98630c4a2e47baef3b3R660
yes